src/main/scala/com/mwz/sonar/scala/scalastyle/ScalastyleQualityProfile.scala

```diff
@@ -51,44 +51,35 @@ object ScalastyleQualityProfile {
   private[scalastyle] final val ProfileName: String = "Scalastyle"
 
   /**
-   * Activates all rules in the Scalastyle rules repository for the given profile.
+   * Activates Scalastyle rules for the given profile excluding blacklisted rules.
+   * Overrides the default severity and parameter values if provided in overrides.
    */
-  def activateAllRules(profile: NewBuiltInQualityProfile): Unit = {
-    ScalastyleInspections.AllInspections
-      .filterNot { i =>
-        ScalastyleRulesRepository.SkipTemplateInstances.contains(i.id) ||
-        ScalastyleRulesRepository.BlacklistRules.contains(i.id)
-      }
-      .foreach(i => profile.activateRule(ScalastyleRulesRepository.RepositoryKey, i.clazz))
-  }
-
-  /**
-   * Enables Scalastyle rules excluding blacklisted rules.
-   * Overrides the default severity and parameter values.
-   */
-  def activateWithOverrides(profile: NewBuiltInQualityProfile, overrides: Overrides): Unit = {
+  def activateRules(profile: NewBuiltInQualityProfile, overrides: Option[Overrides] = None): Unit = {
     ScalastyleInspections.AllInspections
       .filterNot { inspection =>
-        overrides.blacklist.contains(inspection.id) ||
-        inspection.params.nonEmpty
+        ScalastyleRulesRepository.SkipTemplateInstances.contains(inspection.id) ||
+        ScalastyleRulesRepository.BlacklistRules.contains(inspection.id) ||
+        overrides.exists(_.blacklist.contains(inspection.id))
       }
       .foreach { inspection =>
         val rule: NewBuiltInActiveRule =
           profile.activateRule(ScalastyleRulesRepository.RepositoryKey, inspection.clazz)
 
-        // Override the severity.
-        overrides.severities
-          .get(inspection.id)
-          .foreach(severity => rule.overrideSeverity(severity.name))
+        overrides.foreach { overrides =>
+          // Override the severity.
+          overrides.severities
+            .get(inspection.id)
+            .foreach(severity => rule.overrideSeverity(severity.name))
 
-        // Override rule params.
-        overrides.params
-          .get(inspection.id)
-          .foreach {
-            _.foreach {
-              case (k, v) => rule.overrideParam(k, v)
+          // Override rule params.
+          overrides.params
```

Comment:
Maybe two nested **for**s in this case?

Reply:
I'm not sure if having nested `for` comprehensions would make it look any better. I originally had those as two separate `flatMaps` so I could go back to that if it makes it more readable for you?